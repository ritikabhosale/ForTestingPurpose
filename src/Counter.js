import React from 'react';
import {StyleSheet} from 'react-native';
import {Text, View, Button} from 'react-native';

export const Counter = () => {
  const [count, setCount] = React.useState(0);

  return (
    <View>
      <Text style={styles.counterText} testID="counter">
        {count}
      </Text>
      <Button
        testID="incremental-btn"
        onPress={() => setCount(_count => _count + 1)}
        title="press me"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  counterText: {
    color: 'blue',
  },
});
