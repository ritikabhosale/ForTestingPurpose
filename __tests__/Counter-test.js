import 'react-native';
import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {Counter} from '../src/Counter';
import {screen, render} from '@testing-library/react-native';

it('renders correctly', () => {
  render(<Counter />);
});

it('renders with initial count', () => {
  render(<Counter />);
  const counterElement = screen.getByTestId('counter').props;
  expect(counterElement.children).toBe(0);
});

it('renders with incremented count when button pressed', () => {
  render(<Counter />);
  const btnElement = screen.getByTestId('incremental-btn').props;

  act(() => {
    btnElement.onClick();
  });

  const counterElement = screen.getByTestId('counter').props;
  expect(counterElement.children).toBe(1);
});

it('creates snapshot', () => {
  const counterElement = renderer.create(<Counter />);
  expect(counterElement).toMatchSnapshot();
});
