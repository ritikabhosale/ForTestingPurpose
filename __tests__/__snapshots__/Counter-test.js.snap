// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates snapshot 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "blue",
      }
    }
    testID="counter"
  >
    0
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="incremental-btn"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        press me
      </Text>
    </View>
  </View>
</View>
`;
